go-controller/pkg/ovn/master.go

```diff
@@ -485,11 +497,49 @@ func (oc *Controller) ensureNodeLogicalNetwork(nodeName string, hostsubnet *net.
 	return nil
 }
 
+// annotate the node with the subnet information assigned to node's logical switch. the
+// new format of the annotation is:
+//
+// k8s.ovn.org/node-subnets: {
+//	  "default": "192.168.2.1",
+// }
+func (oc *Controller) addNodeAnnotations(node *kapi.Node, subnet string) error {
+	// nothing to do if the node already has the annotation key
+	_, ok := node.Annotations[OvnNodeSubnets]
+	if ok {
+		return nil
+	}
+
+	bytes, err := json.Marshal(map[string]string{
+		"default": subnet,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to marshal node %q annotation for subnet %s",
+			node.Name, subnet)
+	}
+	nodeAnnotations := make(map[string]interface{})
+	// if legacy annotation key exists, then remove it
```

Comment:
@girishmg can we keep the legacy in parallel for a bit?

Reply:
@dcbw do you want to keep both legacy and new annotation for the new nodes that got added after the master was upgraded to the changes in this PR?

None of the worker nodes depend on this annotation, so  they shouldn't see any difference if it is called one way or the other. I think it is better to delete this old annotation now when we have a chance and not carry this legacy thing into future. Please let me know.